src/System.Private.Xml/src/System/Xml/Serialization/Compiler.cs

```diff
@@ -27,9 +27,6 @@ namespace System.Xml.Serialization
 
     internal class Compiler
     {
-#if !XMLSERIALIZERGENERATOR
-        private bool _debugEnabled = DiagnosticsSwitches.KeepTempFiles.Enabled;
-#endif
```

Comment:
@stephentoub `XMLSERIALIZERGENERATOR` is used when we build SGen on core, `Microsoft.XmlSerializer.Generaton` .